source/manual/govuk-site-search.html.md

```diff
@@ -0,0 +1,111 @@
+
+- **User event data pipeline**: BigQuery tables of GA4 user interaction data that are populated via a dataform pipeline, and used by Vertex to train the model serving relevant search results.
+
+The new search stack has a direct dependency on the following existing projects:
+
+- **Finder Frontend**: to display results to users (modified by us to be able to call either the legacy Search API or Search API v2)
+
+- **Search Admin**: Formerly used to configure the legacy search stack, this has been cleaned up to remove legacy functionality but is still used to publish external links (a content type only used in search on GOV.UK) and will be extended in the future to allow configuration of VAIS without requiring developer input
+
+- **Publishing API**: to receive event messages over the publishing message queue (AWS managed RabbitMQ). Note that unlike Search API, there are no direct API calls from or to Whitehall, all content changes go through the MQ
+
+### Core data flows
+
+#### When a content item is published by GOV.UK content editors
+
+- The publishing application pushes the content change to Publishing API
+
+- Publishing API processes the content change and pushes an update message to its RabbitMQ message queue
+
+- Search API v2’s synchronisation worker picks up a change message, acquires a lock on the content ID, extracts content and metadata from the content item based on a set of rules (data paths), and updates the document in VAIS
+
+#### When a user performs a search
+
+- As the user types, autocomplete suggestions are offered by the Search with autocomplete GOV.UK Publishing Components component, through a public proxy endpoint on Finder Frontend that calls the underlying feature on Search API v2
+
+- When the user submits a search request, it is processed by Finder Frontend, which constructs a search query to Search API v2 (according to the same API contract as the legacy Search API)
+
+- Search API v2 receives the query, applies some query-time boosts to it and translates it to a VAIS API call, then maps the API response back to the format expected by Finder Frontend
+
+- Finder Frontend renders the results, together with a VAIS attribution token that is tracked by the analytics code
+
+- Google Analytics (GA4) tracks how the user interacts with results
+
+- GA4 data is processed overnight by the existing GOV.UK analytics pipeline, and ingested into GCP BigQuery.
+
+- Three times a day, the GA4 data is imported and pushed into VAIS as user events. VAIS adjusts the internal ranking model for the engine based on these user events (and some data from its own proprietary search and internet analytics data)
+
+## Infrastructure
+
+### Deployment & environments
+
+The new search stack follows the same environment conventions as other GOV.UK applications, having integration, staging, and production environments. Our GCP account has separate projects for each environment with fully isolated resources.
```

Comment:
Could we add a link to the GCP dev docs page here?
https://docs.publishing.service.gov.uk/manual/google-cloud-platform-gcp.html